### PR TITLE
create a default mediated deposit workflow.

### PR DESCRIPTION
**Pull request by blancoj**
This is to create a default mediated workflow for sufia

### CODE REVIEW

File: lib/generators/sufia/templates/workflow.json.erb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+  "workflows": [
+    {
+      "name": "medidate_depsoit",
+      "label": "Self-deposit workflow",
+      "description": "A multi-step workflow for self-depoist",
+      "actions": [
+         {
+           "name": "finalize_digitization",
```
Comment (jcoyne) [inline lib/generators/sufia/templates/workflow.json.erb]: These don't seem like actions one would want in a mediated deposit workflow.  Am I off base here?

File: lib/generators/sufia/templates/workflow.json.erb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{
+  "workflows": [
+    {
+      "name": "medidate_depsoit",
```
Comment (mjgiarlo) [inline lib/generators/sufia/templates/workflow.json.erb]: Typo. `s/medidate/mediated/`

File: lib/generators/sufia/templates/workflow.json.erb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      "label": "Self-deposit workflow",
```
Comment (mjgiarlo) [inline lib/generators/sufia/templates/workflow.json.erb]: Is this a mediated workflow or a self-deposit workflow?

File: lib/generators/sufia/templates/workflow.json.erb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      "description": "A multi-step workflow for self-depoist",
```
Comment (mjgiarlo) [inline lib/generators/sufia/templates/workflow.json.erb]: Typo. `s/depoist/deposit/`

File: lib/generators/sufia/templates/workflow.json.erb (hunk 1 above)
Comment (mjgiarlo) [inline lib/generators/sufia/templates/workflow.json.erb]: I wondered the same.

**Comment by mjgiarlo** [comment]
Travis is :red_circle:.

File: lib/generators/sufia/templates/workflow.json.erb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+           "name": "complete",
+           "from_states": [{"names": ["final_review"], "roles": ["completing"]}],
+           "transition_to": "complete",
+           "attributes": {
+             "presentation_sequence": 3
+           },
+           "notifications": [
+             {
+               "notification_type": "email",
+               "name": "Sufia::Workflow::ConfirmSubmission",
```
Comment (mjgiarlo) [inline lib/generators/sufia/templates/workflow.json.erb]: This line will cause a failure unless `Sufia::Workflow::ConfirmSubmission` is a real thing. Remove notifications block from JSON.

File: lib/generators/sufia/templates/workflow.json.erb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      "label": "Mediated deposit workflow",
+      "description": "A multi-step workflow for mediated deposit",
+      "actions": [
+         {
+           "name": "deposit",
+           "from_states": [],
+           "transition_to": "pending"
+         },
+         {
+           "name": "finalize_metadata",
```
Comment (jcoyne) [inline lib/generators/sufia/templates/workflow.json.erb]: Get rid of this action.

File: lib/generators/sufia/templates/workflow.json.erb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+           "from_states": [{"names": ["pending"], "roles": ["finalizing"]}],
+           "transition_to": "final_review",
+           "attributes": {
+                "presentation_sequence": 1
+           }
+         }, {
+           "name": "complete",
```
Comment (jcoyne) [inline lib/generators/sufia/templates/workflow.json.erb]: probably change this to `review`

Comment (mjgiarlo) [inline lib/generators/sufia/templates/workflow.json.erb]: I suggest an even simpler one-step approval workflow: https://gist.github.com/mjgiarlo/f8d1ea8e14f549f3513081e2e62ddb52

**Comment by mjgiarlo** [comment]
Working on https://github.com/projecthydra/curation_concerns/pull/1089 to make sure we have a viable default workflow in place in addition to the new 1-step workflow. Once that's merged and this branch picks up that work, we should wire this 1-step workflow into the current admin set WF strategy.

**Comment by mjgiarlo** [comment]
Two steps to integrate the new workflow with the existing AS strategy:

1. When a `PermissionTemplate` is created, set its `workflow_name` to `one_step_mediated_deposit`
2. Pass in attrs that include `{ admin_set_id: foo }` where `foo` is the id of the AS to which the `PermissionTemplate` in step 1 is attached.

**Pull request merged**